components/com_contact/src/Model/ContactModel.php

```
@@ -300,8 +300,16 @@ public function getItem($pk = null)
 			}
 			catch (\Exception $e)
 			{
-				$this->setError($e);
-				$this->_item[$pk] = false;
+				if ($e->getCode() == 404)
+				{
+					// Need to go through the error handler to allow Redirect to work.
+					throw new \Exception($e->getMessage(), 404);
```

Can we re-throw the original exception rather than creating a fresh one. At worse we should nest the existing exception as the third param

---

@wilsonge 
That code is the same as in ArticleModel.php. Should the eventual modification be done for both in another PR in order to get that in already for beta4?

---

Sure I'm happy. But we definitely should at minimum nest the exceptions

---

@wilsonge did you mean `throw $e;`  ?

---

Ideally yes. Else if that doesn't work for some reason - worse case change this line to

`throw new \Exception($e->getMessage(), 404, $e);`

---

ah ok .... `throw $e;` works 
i'll do a patch with that for both ArticleModel and ContactModel

---

see #30653